### ISSUE TITLE
Use of example mysql tables and config in Verdaccio 5.0.3 fails - unhandled exception

### ISSUE THREAD

**Issue by ghost** [bchanudet/verdaccio-mysql#1]
This is two problems, firstly the exception is unhandled which means the server crashes and secondly the example queries provided either fail or cause failure in the associated JS.

The trigger the error, have a an empty db based on the example sql at https://github.com/bchanudet/verdaccio-mysql/blob/master/example/mysql-database.sql

issue an npm command:
```
npm adduser --registry http://npm.example.com/
```

The process does not add any user_groups so it seems that the SQL which is run to return them means usergroups is set to NULL.

```SELECT u.`username`, GROUP_CONCAT(g.`name`) AS usergroups FROM `users` u LEFT JOIN `user_group` ug ON ug.`username` = u.`username` LEFT JOIN `groups` g ON ug.`group` = g.`name` WHERE u.`username` = ? AND u.`password` = PASSWORD(?)```

![image](https://user-images.githubusercontent.com/78358451/116870279-2bd59480-ac0a-11eb-987a-0b82d2411622.png)

Do we need to add user_groups before we can add a user?

**Verdaccio 5.0.3 logs**
```
 http --- 404, user: null(192.168.88.254 via undefined), req: 'POST /-/v1/login', bytes: 20/150
  express:router dispatching PUT /-/user/org.couchdb.user:tshaw +8s
  express:router query  : /-/user/org.couchdb.user:tshaw +0ms
  express:router expressInit  : /-/user/org.couchdb.user:tshaw +1ms
  express:router corsMiddleware  : /-/user/org.couchdb.user:tshaw +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:tshaw +0ms
 http --- undefined requested 'PUT /-/user/org.couchdb.user:tshaw'
  express:router errorReportingMiddleware  : /-/user/org.couchdb.user:tshaw +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:tshaw +0ms
  express:router compression  : /-/user/org.couchdb.user:tshaw +0ms
  express:router router  : /-/user/org.couchdb.user:tshaw +1ms
  express:router dispatching PUT /-/user/org.couchdb.user:tshaw +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:tshaw +0ms
  express:router jsonParser  : /-/user/org.couchdb.user:tshaw +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:tshaw +1ms
  express:router encodeScopePackage  : /-/user/org.couchdb.user:tshaw +0ms
 fatal--- uncaught exception, please report this
TypeError: Cannot read property 'usergroups' of undefined
    at Query.<anonymous> (/usr/lib/node_modules/verdaccio-mysql/lib/index.js:56:36)
    at Query.<anonymous> (/usr/lib/node_modules/verdaccio-mysql/node_modules/mysql/lib/Connection.js:526:10)
    at Query._callback (/usr/lib/node_modules/verdaccio-mysql/node_modules/mysql/lib/Connection.js:488:16)
    at Query.Sequence.end (/usr/lib/node_modules/verdaccio-mysql/node_modules/mysql/lib/protocol/sequences/Sequence.js:83:24)
    at Query._handleFinalResultPacket (/usr/lib/node_modules/verdaccio-mysql/node_modules/mysql/lib/protocol/sequences/Query.js:149:8)
    at Query.OkPacket (/usr/lib/node_modules/verdaccio-mysql/node_modules/mysql/lib/protocol/sequences/Query.js:74:10)
    at Protocol._parsePacket (/usr/lib/node_modules/verdaccio-mysql/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/usr/lib/node_modules/verdaccio-mysql/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/usr/lib/node_modules/verdaccio-mysql/node_modules/mysql/lib/protocol/Parser.js:43:10)
    at Protocol.write (/usr/lib/node_modules/verdaccio-mysql/node_modules/mysql/lib/protocol/Protocol.js:38:16)
```

**Verdaccio config**
```
auth:
    mysql:
        connection:
            host: "127.0.0.1"
            port: 3306
            user: "verdaccio"
            password: "password"
            database: "verdaccio"
        queries:
            add_user: "INSERT INTO `users`(`username`,`password`) VALUES (?,PASSWORD(?));"
            update_user: "UPDATE `users` SET `password`= PASSWORD(?) WHERE `username`=? AND `password`=PASSWORD(?);"
            auth_user: "SELECT u.`username`, GROUP_CONCAT(g.`name`) AS usergroups FROM `users` u LEFT JOIN `user_group` ug ON ug.`username` = u.`username` LEFT JOIN `groups` g ON ug.`group` = g.`name` WHERE u.`username` = ? AND u.`password` = PASSWORD(?)"
```

**Comment by bchanudet** [comment]
Seems like a strange one, the checks done there should have detected it in some way: https://github.com/bchanudet/verdaccio-mysql/blob/cedb0fb8e1531fd184f4edbb3d07ef721af96f42/src/index.ts#L53 

The only way for me for these conditions to pass would be that `result = [undefined]` but I just don't see how it can happen without `node-mysql` triggering an error. 

Could you try adjusting the query to avoid sending back `null` as usergroups?  Something along this: 
```sql
SELECT u.`username`, COALESCE(GROUP_CONCAT(g.`name`),'') AS usergroups FROM `users` u LEFT JOIN `user_group` ug ON ug.`username` = u.`username` LEFT JOIN `groups` g ON ug.`group` = g.`name` WHERE u.`username` = ? AND u.`password` = PASSWORD(?);
```

Anyway I'll have a look and see if I can reproduce it on my side. I'll also take the time to fix #2 :)

**Comment by bchanudet** [comment]
Alright I had a closer look this evening. Looks like my testing process was a bit too light at the time. :) 

I fixed most of the issues in 6278dffcf68103f5759d2f132d197fecc1194fb8. I also did a bit of work to improve the output to the console when the query doesn't return the expected columns from the database. 

The package in the NPM registry is already updated to the 1.0.1 version. :)
